checkov/terraform/checks/graph_checks/gcp/GCPPostgreSQLDatabaseFlaglog_executor_statsIsSetToOFF.yaml

```diff
@@ -0,0 +1,23 @@
+metadata:
+  name: "Ensure PostgreSQL database flag 'log_executor_stats' is set to 'off'"
+  id: "CKV2_GCP_14"
+  category: "GENERAL"
+definition:
+  and:
+    - cond_type: "attribute"
+      resource_types:
+        - "google_sql_database_instance"
+      attribute: "database_version"
+      operator: "contains"
+      value: "POSTGRES"
+    - or:
+        - cond_type: "attribute"
+          resource_types:
+            - "google_sql_database_instance"
+          attribute: "settings.database_flags[*]"
+          operator: "jsonpath_not_exists"
+        - cond_type: "attribute"
+          resource_types:
+            - "google_sql_database_instance"
+          attribute: "settings.database_flags[?(@.name == log_executor_stats & @.value == on)]"
+          operator: "jsonpath_not_exists"
```

Comment:
this should be similar to `CKV2_GCP_13` one `or:` and then it is either not a postgresql database or any of the jsonpaths can be found.

Reply:
CKV2_GCP_13 - has one big `or:` and the rest 4 Database policies has `and:`  & `or:` because,

By default, GCP turns **off** these flags, so these flags will be turned off when not used in the script.
https://cloud.google.com/sql/docs/postgres/flags

So we have three cases to be satisfied,

1. It should be a PostgreSQL instance
2. The corresponding flag attribute should not be part of the settings
3. If the flag is present, then it should be turned OFF

That's why I have mentioned `or:` & `and:` for the policies where flags need to be turned **OFF**
Let me know if you have a different logic.

Reply:
the problem is with your current definition is, any `google_sql_database_instance` which is not PostgreSQL will fail, just add a MySQL example into the test file of this and the other checks and then you will see that they fail. That's why you need to change the check definition

Reply:
Hi Anton, thanks for the help, I missed testing for MySQL example to the test file.
I have modified the policy as advised.
Please check & let me know if there are any issues.